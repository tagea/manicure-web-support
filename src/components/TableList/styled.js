import styled from 'styled-components';
import Paper from '@material-ui/core/Paper';
import TableCell from '@material-ui/core/TableCell';
import Grid from '@material-ui/core/Grid';

import { Table } from 'react-virtualized';
import { theme } from '../../ui/Themes';

const { palette } = theme;

export const PaperStyled = styled(Paper)`
  &&{
    background-color: ${palette.surface.light};
    border-radius: 3px;
    box-shadow: none;
    height: 100%;
  }
`;

export const TableStyled = styled(Table)`
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  box-sizing: border-box;
  div:focus{
      outline: none;
  }
  .rowTable{
    display: flex;
    padding: 30px 25px;
    box-sizing: border-box;
    border-bottom: 1px solid ${palette.border.light};
    border-top: 1px solid transparent;
    transition: all 0.2s ease-in;
    :hover{
      background-color: ${palette.surface.transparent};
      border-bottom: 1px solid ${palette.border.main};
      border-top: 1px solid ${palette.primary.main};
    }
  }
  .ReactVirtualized__Table__headerRow:hover{
    background-color: transparent;
    border-bottom: 1px solid ${palette.border.light};
    border-top: 1px solid transparent;
    
  }
  .headerStyle{
    padding-top: 0;
    padding-bottom: 0;
    line-height: 20px;
    color: ${palette.text.transparent};
  }
`;

export const TableCellStyled = styled(TableCell)`
  &&{
    font-size: 1.1em;
    border-bottom: none;
    padding: 0px;
  }
`;

export const TableHeader = styled(Grid)` 
  &&{
    display:flex;
    justify-content: space-between;
    margin: 0 20px;
    padding-top: 25px;
  }
`;
