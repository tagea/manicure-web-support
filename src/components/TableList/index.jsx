import React from 'react';
import PropTypes from 'prop-types';
import { InfiniteLoader, AutoSizer, Column } from 'react-virtualized';
import { FabButton } from '../../ui/FabButton';
import { SearchInput } from '../../ui/SearchInput';
import { PaperStyled, TableHeader, TableStyled, TableCellStyled } from './styled';
import AddSrc from '../../img/plus.svg';

class TableList extends React.Component {
  render() {
    const { loadMoreRows, listData, columns, rowCount, handleOpen } = this.props;
    return (
      <PaperStyled>
        <TableHeader>
          <SearchInput />
          <FabButton label="Add" src={AddSrc} handleClick={handleOpen} />
        </TableHeader>

        <InfiniteLoader
          isRowLoaded={({ index }) => listData[index]}
          loadMoreRows={loadMoreRows}
          rowCount={rowCount} // all list length (all data)
        >
          {({ onRowsRendered, registerChild }) => (
            <AutoSizer>
              {({ width, height }) => (
                <TableStyled
                  ref={registerChild}
                  onRowsRendered={onRowsRendered}
                  height={height - 220}
                  width={width}
                  defaultWidth={width}
                  rowCount={listData.length}
                  rowGetter={({ index }) => listData[index]}
                  rowHeight={84}
                  columns={columns}
                  headerClassName="headerStyle"
                  rowClassName="rowTable"
                >
                  {columns.map((headerName, index) => (
                    <Column
                      key={index}
                      dataKey={headerName.dataKey}
                      headerRenderer={() => (
                        <TableCellStyled component="div">{headerName.label}</TableCellStyled>
                      )}
                      width={headerName.width}
                      flexGrow={headerName.flexGrow}
                    />
                  ))}
                </TableStyled>
              )}
            </AutoSizer>
          )}
        </InfiniteLoader>
      </PaperStyled>
    );
  }
}

TableList.propTypes = {
  loadMoreRows: PropTypes.func,
  listData: PropTypes.instanceOf(Array),
  columns: PropTypes.arrayOf(PropTypes.object),
  rowCount: PropTypes.number,
  handleOpen: PropTypes.func
};

TableList.defaultProps = {
  loadMoreRows: () => false,
  listData: [],
  columns: [{}],
  rowCount: 10000,
  handleOpen: () => false
};

export { TableList };
