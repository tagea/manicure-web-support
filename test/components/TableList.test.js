import React from 'react';
import { shallow } from 'enzyme';
import { InfiniteLoader } from 'react-virtualized';
import { TableList } from '../../src/components/TableList/index';

it('render TableList component', () => {
  const TableListComponent = shallow(<TableList />);
  expect(TableListComponent).toMatchSnapshot();
});

it('InfinitLoader should contain "minimumBatchSize" property with "10" value', () => {
  const TableListComponent = shallow(<TableList />);
  expect(TableListComponent.find(InfiniteLoader).prop('minimumBatchSize')).toBe(10);
});

it('InfinitLoader should contain "rowCount" property with "10000" value', () => {
  const TableListComponent = shallow(<TableList />);
  expect(TableListComponent.find(InfiniteLoader).prop('rowCount')).toBe(10000);
});

it('InfinitLoader should contain "threshold" property with "15" value', () => {
  const TableListComponent = shallow(<TableList />);
  expect(TableListComponent.find(InfiniteLoader).prop('threshold')).toBe(15);
});

it('InfiniteLoader should have isRowLoaded function', () => {
  const TableListComponent = shallow(<TableList />);
  expect(TableListComponent.find('InfiniteLoader').prop('isRowLoaded')).toBeInstanceOf(Function);
});

it('InfiniteLoader should have loadMoreRows function', () => {
  const TableListComponent = shallow(<TableList />);
  expect(TableListComponent.find('InfiniteLoader').prop('loadMoreRows')).toBeInstanceOf(Function);
});
