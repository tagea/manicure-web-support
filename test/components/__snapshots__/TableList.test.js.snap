// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render TableList component 1`] = `
<styled__PaperStyled>
  <styled__TableHeader>
    <SearchInput />
    <FabButton
      handleClick={[Function]}
      label="Add"
      src="test-file-stub"
    />
  </styled__TableHeader>
  <InfiniteLoader
    isRowLoaded={[Function]}
    loadMoreRows={[Function]}
    minimumBatchSize={10}
    rowCount={10000}
    threshold={15}
  >
    <Component />
  </InfiniteLoader>
</styled__PaperStyled>
`;
